File: app/src/main/java/com/jacksonke/teresapassword/DBHelper.java
package com.jacksonke.teresapassword;

import android.content.Context;
import android.support.annotation.NonNull;

import java.util.List;

import io.objectbox.Box;
import io.objectbox.BoxStore;

public class DBHelper {

    private static DBHelper sInstance = new DBHelper();

    private BoxStore boxStore;

    private boolean isInited = false;

    private DBHelper() {

    }

    public static DBHelper getInstance(){
        return sInstance;
    }

    public void init(Context context){
        if (isInited){
            throw new RuntimeException("boxStore has alwready been built");
        }

        boxStore = MyObjectBox.builder().androidContext(context).build();
    }

    public BoxStore getBoxStore() {
        return boxStore;
    }

    public List<SiteEntity> queryAll(){
      Box<SiteEntity> box = DBHelper.getInstance().getBoxStore().boxFor(SiteEntity.class);
      return box.getAll();
    }

    public SiteEntity queryBySite(String site){
      Box<SiteEntity> box = DBHelper.getInstance().getBoxStore().boxFor(SiteEntity.class);
      return box.query().equal(SiteEntity_.name, site).build().findFirst();
    }

    public long insertSiteEntity(@NonNull SiteEntity entity){
      assert entity.id == 0;
      Box<SiteEntity> box = DBHelper.getInstance().getBoxStore().boxFor(SiteEntity.class);
      return box.put(entity);
    }

    public void updateSiteEntity(@NonNull SiteEntity entity){
      assert entity.id != 0;
      Box<SiteEntity> box = DBHelper.getInstance().getBoxStore().boxFor(SiteEntity.class);
      box.put(entity);
    }

    public void rmSiteEntity(String siteName){
      Box<SiteEntity> box = DBHelper.getInstance().getBoxStore().boxFor(SiteEntity.class);
      List<SiteEntity> siteEntities = box.query().equal(SiteEntity_.name, siteName).build().find();
      SiteEntity item = null;
      if (siteEntities.size() > 0){
        item = siteEntities.get(0);
      }

      if (item != null){
        box.remove(item);
      }
    }


}

File: app/src/main/java/com/jacksonke/teresapassword/DBHelper.kt
package com.jacksonke.teresapassword

import android.content.Context
import io.objectbox.BoxStore
import io.objectbox.query.QueryBuilder

class DBHelper private constructor() {
    private var boxStore: BoxStore? = null
        private set
    private val isInited = false
    fun init(context: Context?) {
        if (isInited) {
            throw RuntimeException("boxStore has already been built")
        }
        boxStore = MyObjectBox.builder().androidContext(context).build()
    }

    fun queryAll(): List<SiteEntity> {
        val box = instance.boxStore!!.boxFor(
            SiteEntity::class.java
        )
        return box.all
    }

    fun queryBySite(site: String): SiteEntity? {
        val box = instance.boxStore!!.boxFor(
            SiteEntity::class.java
        )
        return box.query().equal(SiteEntity_.name, site, QueryBuilder.StringOrder.CASE_SENSITIVE).build().findFirst()
    }

    fun insertSiteEntity(entity: SiteEntity): Long {
        assert(entity.id == 0L)
        val box = instance.boxStore!!.boxFor(
            SiteEntity::class.java
        )
        return box.put(entity)
    }

    fun updateSiteEntity(entity: SiteEntity) {
        assert(entity.id != 0L)
        val box = instance.boxStore!!.boxFor(
            SiteEntity::class.java
        )
        box.put(entity)
    }

    fun rmSiteEntity(siteName: String) {
        val box = instance.boxStore!!.boxFor(
            SiteEntity::class.java
        )
        val siteEntities: List<SiteEntity> =
            box.query().equal(SiteEntity_.name, siteName, QueryBuilder.StringOrder.CASE_SENSITIVE).build().find()
        var item: SiteEntity? = null
        if (siteEntities.isNotEmpty()) {
            item = siteEntities[0]
        }
        if (item != null) {
            box.remove(item)
        }
    }

    companion object {
        val instance = DBHelper()
    }
}